src/Utils/arm/RbacUtils.ts

```diff
@@ -0,0 +1,120 @@
+import { configContext } from "ConfigContext";
+import { armRequest } from "Utils/arm/request";
+import { getCopyJobAuthorizationHeader } from "../CopyJobAuthUtils";
+
+export type FetchAccountDetailsParams = {
+  subscriptionId: string;
+  resourceGroupName: string;
+  accountName: string;
+};
+
+export type RoleAssignmentPropertiesType = {
+  roleDefinitionId: string;
+  principalId: string;
+  scope: string;
+};
+
+export type RoleAssignmentType = {
+  id: string;
+  name: string;
+  properties: RoleAssignmentPropertiesType;
+  type: string;
+};
+
+type RoleDefinitionDataActions = {
+  dataActions: string[];
+};
+
+export type RoleDefinitionType = {
+  assignableScopes: string[];
+  id: string;
+  name: string;
+  permissions: RoleDefinitionDataActions[];
+  resourceGroup: string;
+  roleName: string;
+  type: string;
+  typePropertiesType: string;
+};
+
+const apiVersion = "2025-04-15";
+
+const getArmBaseUrl = (): string => {
+  const base = configContext.ARM_ENDPOINT;
+  return base.endsWith("/") ? base.slice(0, -1) : base;
+};
+
+const buildArmUrl = (path: string): string => `${getArmBaseUrl()}${path}?api-version=${apiVersion}`;
```

Comment:
Probably useful for more than just RBAC. I would think about moving these (and any other common ARM-related functions) to its own file. I'm actually surprised these methods hadn't already been written. Thank you for doing it!